### DISCUSSION TITLE
Open Pedalboard Platform

### DISCUSSION THREAD

**Discussion by laenzlinger** [shanteacontrols/OpenDeck#322] [General]
Hi 

I am working for a while on a similar project: https://github.com/pedalboard

Part of the project is a midi controller, which is based on specific firmware written in Rust.

Now i want to make this firmware more generic. If possible, I would like to reuse your firmware and create a custom configuration for my hardware. 

But my hardware has some specific feature, which do not seem to be supported by your firmware:

* Special LED output (LED rings) based addressable RGB LEDs.
* Two OLED screens (SSD-1327), which I would like to show some configuration values for each button.
 
Therefore I would probably need to extend your firmware to support this specific hardware. How are you storing the configuration? Are you supporting an eeprom in your firmware?

An alternative would be to implement your configuration protocol on my firmware (instead of my own). 

Are you interested in a collaboration?

You can contact me on Discord https://discord.gg/ncyKyryHAc

Christof

**Comment by laenzlinger** [comment]
I have started to implement the protocol in rust: https://github.com/pedalboard/opendeck

Its of course in a very prototype state. I also found some inconsistencies between your UI implementation and your docu in the wiki

**Comment by laenzlinger** [comment]
BTW: My goal is in now way to compete with your platform. I want to reuse your platform in my project which focuses on audio processing but requires some IO, where I want to reuse parts of your platform.

**Comment by paradajz** [comment]
> Therefore I would probably need to extend your firmware to support this specific hardware. How are you storing the configuration? Are you supporting an eeprom in your firmware?

I'm not sure what exactly are you interested in here. I'm using several layers of abstraction so that the final configuration layout is hardware independent.

> I also found some inconsistencies between your UI implementation and your docu in the wiki

Which ones?

**Comment by laenzlinger** [comment]
An example of such an inconsistency: Encoder list of section only lists 8 values but there are for more values configurable

**Comment by laenzlinger** [comment]
A BPM value can be modified with a button or an encode. However, by reading the docs I could not find out how the BPM value is used.

**Comment by paradajz** [comment]
I'm sorry for the very late reply but just wanted to mention that documentation in general has been updated.
